app/src/main/java/com/android/sample/model/profile/ProfilesRepositoryFirestore.kt

```diff
@@ -72,6 +75,48 @@ open class ProfilesRepositoryFirestore @Inject constructor(private val db: Fireb
         }
   }
 
+  override fun addLikedActivity(
+      userId: String,
+      activityId: String,
+      onSuccess: () -> Unit,
+      onFailure: (Exception) -> Unit
+  ) {
+    db.collection("profiles")
+        .document(userId)
+        .update("likedActivities", FieldValue.arrayUnion(activityId))
+        .addOnCompleteListener { task ->
+          if (task.isSuccessful) {
+            onSuccess()
+          } else {
+            task.exception?.let { e ->
+              Log.e("ProfilesRepository", "Error adding activity to profile", e)
+              onFailure(e)
+            }
+          }
+        }
+  }
+
+  override fun removeLikedActivity(
+      userId: String,
+      activityId: String,
+      onSuccess: () -> Unit,
+      onFailure: (Exception) -> Unit
+  ) {
+    db.collection("profiles")
+        .document(userId)
+        .update("likedActivities", FieldValue.arrayRemove(activityId))
+        .addOnCompleteListener { task ->
+          if (task.isSuccessful) {
+            onSuccess()
+          } else {
+            task.exception?.let { e ->
+              Log.e("ProfilesRepository", "Error adding activity to profile", e)
```

Comment:
The error log for removeLikedActivity incorrectly mentions "Error adding activity to profile".